### ISSUE TITLE
No-Fall preventing real/legit damage

### ISSUE THREAD

**Issue by PT400C** [NoCheatPlus/Issues#485]
#### Short description of the issue.
The anti-nofall module does prevent falldamage which should the players get.
(Closer description in the screenshot)

#### Complete output of the _ncp version_ command (please prefer text here).

---- Version information ----
#### Server ####
git-Paper-773 MC: 1.9.4
detected: 1.9.4
#### NoCheatPlus ####
Plugin: 3.15.2-SNAPSHOT-sMD5NET-b1113
MCAccess: 1.9.4 / Spigot-CB-1.9_R2
Features:
blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8 | BlocksMC1_9
checks: FastConsume | Gutenberg | HotFixFallingBlockPortalEnter | AttackFrequency | FlyingFrequency | KeepAliveFrequency
defaults: pvpKnockBackVelocity
packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance
Hooks: AllViolationsNCP 1.0
#### Related Plugins ####
ProtocolLib v4.3.0 | ViaVersion v1.1.1

#### How to reproduce the issue (is it random / always happens / side conditions)?
It is always happening. Just do everything described in this screenshot:
[Example setup](https://img1.picload.org/image/rwlarwli/2017-08-16_08.31.10.png)

#### Have you made any (noteworthy) changes to the default configuration of NCP?
None really breaking changes and no changes to the no-fall section.

#### Do you run any other special plugins that affect game mechanics next to NC+ (adding skills, machines, adding/changing blocks, other anti cheating)?
none

**Comment by Janmm14** [comment]
Is it unrelated to the used client version?

**Comment by PT400C** [comment]
Yeah :/

**Comment by asofold** [comment]
Is/was the piston moving, so you could've touched the extended piston in theory (not necessarily actually)?

If yes, perhaps NCP detected "lost ground" with touching the piston, because NCP can't yet judge if it should or not assume ground in such a case, so it always would.

If not... will have to reproduce. Mind a debug log always being the door opener for fastest fixes: https://github.com/NoCheatPlus/Docs/wiki/Debugging#on-the-fly-debug-output-for-individual-players

**Comment by asofold** [comment]
Uncertain if it's this - 3.5 fall distance will lead to damage from build 1146 on.

Concerning pistons ... it might happen in future, that we assume falling instead of 'opportunistic on-ground', if the parameters resemble falling with friction (including hitting ground below the past piston move), but it's not in focus right now.

Closing as 'postponed'.
